apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/commons/Context.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.solrj.commons;
+
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+
+public class Context {
+    private static final ThreadLocal<Context> CONTEXT = new ThreadLocal<Context>() {
```

Comment:
Don't use your own threadlocal. Use RunningContext if you need this.

Reply:
And only use RunningContext when dynamicField can't support the scenario. Because anything in runningContext or ThreadLocal has leak risk.

Reply:
DynamicField cannot used in multi-thread context. It is not thread-safe.
I have some variables need to store in thread context.
If I have to store things in thread context, I can use RunningContext. Right?

Reply:
Yes, if the object is used in multiple threads, then dynamic field doesn't fit. Please use RunningContext.

Reply:
I got it

Reply:
I noticed that the `Context` you defined is used only for setting the tags? In the documentation,   [Advanced API's](https://github.com/apache/skywalking/blob/master/docs/en/guides/Java-Plugin-Development-Guide.md#advanced-apis), there is a paragraph saying:

> There is a set of advanced APIs in Span, which work specific for async scenario. When tags, logs, attributes(**including end time**) of the span needs to set in another thread, you should use these APIs.

seems these API's are more suitable here?

Reply:
**RunningContext** designed for in thread propagation to avoid one object accessed by multiple threads. **Async APIs**  designed for the span across threads to finish and set tag.

Choose the right API fitting your scenario.

Reply:
Thanks for clarifying this